### DISCUSSION TITLE
Special layout for matrix fields with only one row

### DISCUSSION THREAD

**Discussion by acalvino4** [craftcms/cms#10188] [Ideas]
When a matrix field is limited to one row, it effectively becomes a switcher for different block types. A key example of this is a link field:
![image](https://user-images.githubusercontent.com/43254861/144092472-91e40fc0-70cb-43ca-a3fb-8a0bdcb13baa.png)
![image](https://user-images.githubusercontent.com/43254861/144092660-54d7d5b7-b0f5-4a93-99c4-1e13b605e574.png)

While this interface is **_great_** for a field with multiple rows, I see three things that make it unideal UX for a matrix field of only one row:
1. `Page` (in my example snapshots) may be the most common case, but it takes an extra click to get there.
2. In the second snapshot, it is not the most obvious how to switch the link type to `External`, and even once you figure out you need to _delete_, then _add_, it is extra clicks to first delete the `Page` block, and then add the `External` block.
3. Those greyed out boxes at the bottom of the second snapshot are completely irrelevant and just clutter up the interface

Now, I found the "Typed Link" plugin which handles all these issues beautifully:
![image](https://user-images.githubusercontent.com/43254861/144094159-d5745c30-1cf0-4749-9182-189d9904df93.png)

Switching is as simple as navigating the dropdown, the ui is intuitive, you can set a default, and the interface is less cluttered.

## My proposal

Check out this toggle in the form for creating a super table:
![image](https://user-images.githubusercontent.com/43254861/144094800-4193407e-d15f-4fda-ad3e-2bb39eade998.png)

I want to propose adding a similar toggle to the creation of matrix fields (and who knows, possibly the native table fields), which then optimized the layout for that type of field.

This would enable users to easily create something like the "typed link" field for a variety of use cases and would streamline the backend editing experience.

Curious to know others' thoughts!

**Comment by brandonkelly** [comment]
Related: #9384, #8602
